impl/client/src/main/javascript/web/repo/state.js

```diff
@@ -263,11 +263,11 @@ pentaho.userSettings.prototype.generateUniqueUrl = function(url) {
 pentaho.userSettings.prototype.getSettings = function( names, callback, caller ) {
 
   dojo.xhrGet({
-    url: this.generateUniqueUrl(CONTEXT_PATH + 'content/ws-run/UserSettingService/getUserSettingsJson'),
```

Comment:
Are we sure this method is only used by EE installations, and, moreover, with PIR installed?

Reply:
Yeah, I see the uses here, for `pentaho.userSettingsInstance`: https://github.com/search?q=org%3Apentaho%20pentaho.userSettingsInstance&type=code.

The Java `UserSettingService` class was duplicated in PIR and Common-UI. However, the client JS code only exists in Common-UI. I found no consumers of PIR's own REST endpoint provided by that class, at `(...)/pentaho-interactive-reporting/api/UserSettingService`. Wasn't it possible to keep the Common-UI copy/version of this class instead?

Reply:
I guess it boils down to the question of whether we want a service used by exactly one plugin to live in a "common" repo or the plugin itself.  At the time I worked on this, I was leaning towards the latter, and I contemplated moving this state.js module to PIR as well.  You'll also note that the contents of the web/dataapi folder are duplicated between here and the data-access repo: https://github.com/pentaho/data-access/tree/master/assemblies/data-access-plugin/src/main/resources/resources/web.  This led me to believe (possibly incorrectly) that at one time we had most things in common-ui and started (and didn't finish) migrating them to separate plugins.

Reply:
The Java service and the single JS code consuming it are now in two different repositories, one CE consuming another EE.

I think it would be best to default to _preserving compatibility_ to a (for all we know) public API (possibly being used by a third-party). The JS class would be kept in Common-UI (as it has been in this PR). Then, the Common-UI endpoint would be converted to JAX-RS (by copying PIR's). Optionally, PIR's endpoint could be removed, given, it's not used internally, nor, afaict, should it be considered a public API.

If not preserving compatibility (including for CE users), then I think it would be best to move the JS class to PIR as well.

Either solution would be "cohesive", even though I lean towards the first.

Not that it matters that much, but I believe the Common-UI code was an attempt to centralize pieces of code being shared/developed by/for the EE plugins (Analyzer, PIR). This includes Dojo UI pieces and other APIs such as the "Data API". Some only used by Analyzer, others only by PIR. Still, I believe the goal was to have a "common" UI library that could be used not only by our plugins, but also by third-party ones.